### PR TITLE
Add support for absolute paths.

### PR DESCRIPTION
**Pull request by paulmillr**
Sometimes it’s pretty useful.

### CODE REVIEW

**Comment by juliangruber** [comment]
+1

**Comment by tj** [comment]
what's the use-case(s)? -1 from me in general, I'd consider that an anti-pattern since it's effectively bypassing our well-defined dependencies

**Comment by paulmillr** [comment]
Simplifying of transitioning to relative paths. I want to use the require in [brunch](http://brunch.io) since I plan to integrate **component-builder** stuff and make awesome cross-language builder.

In [example app](https://github.com/paulmillr/ostio/tree/master/app) you can see `require`-s all over the place. The app will require a lot of refactoring for working with the require definition.

With absolute path support, it only requires to change `require 'stuff'` to `require '/ostio/app/stuff'` etc.

With this, a lot of brunch users will be able to start using and creating components right in the next release, this week. But of course I will add warning that absolute paths should not be used and exist there only for transition.

**Comment by tj** [comment]
that's a large reason why I dont want to go that route with component personally, I'll probably merge for parity with node but I think this is an anti-pattern, large rails-style app trees are really difficult to work with, you just end up with a ton of `../../..../.../` and second-class files with no identity. IMO if you cant have a flat list of files that have meaningful names they shouldn't exist in the first place. Also treating the entire app as a component forces all the dependencies to be defined out of context, there's pros and cons to this part but I think in general that's an anti-pattern, im not sure why everyone modularizes until it gets to the app level

File: test/require.js [hunk 1]
```diff
@@ -94,6 +94,12 @@ describe('require.register(name, fn)', function(){
     var ret = eval(js + 'require("toString")', {});
     ret.stuff.should.equal(228);
   })
+
+  it('should support absolute paths', function() {
+    var js = fixture('absolute.js');
+    var ret = eval(js + 'require("meh")', {});
```
Comment (tj) [inline test/require.js:100]: I dont think we should treat ones without slashes as absolute, it looks like a faulty `./deps/NAME` lookup IMO

**Comment by paulmillr** [comment]
Also, I totally agree, apps should be built from simple modular components without deep file tree and jquery-240k-bullshit.

I plan to do my next big app with components all over the place, the absolute paths stuff will only make transitioning to it easier.

File: test/require.js (hunk 1 above)
Comment (paulmillr) [inline test/require.js:100]: they ain’t treated as absolute, default behavior wasn’t changed. I just forgot to use `/meh` here. it works this way anyway.

Comment (tj) [inline test/require.js:100]: I guess node kind of allows this as well, not sure it should though haha, weird "feature" to be picking out random modules from all over, I'll merge and just leave it undocumented though

Comment (paulmillr) [inline test/require.js:100]: :+1: 

would you please push new release of builder with this and async hooks?

Comment (tj) [inline test/require.js:100]: ah nvm i see that was just the entry point, I thought it was facilitating node-style `require('jade')` from within something that does not define it as a dep (weird stuff)

**Pull request merged**